libcontainerd/client_daemon.go

```diff
@@ -213,6 +215,29 @@ func (c *client) Restore(ctx context.Context, id string, attachStdio StdioCallba
 	return alive, pid, nil
 }
 
+// The problem with this approach is the split logic between the plugin-like code in daemon/gpu/nvidia
+// and the containerd client code which needs to embed the GPU API calls directly.
+func setGPUOptions(ociOpts []oci.SpecOpts, ociSpec *specs.Spec) []oci.SpecOpts {
```

Comment:
Just call this code in the handler.  You can call the spec opts like any other function.  You can just make a simple combinator for this.

in your gpu handler dont use annotations and just process the fields like:

```go
func applyOpts(s *specs.Spec, opts ...SpecOpts) {
    for _, o := range opts {
         o(ctx, nil, nil, s)
    }
}
```

Reply:
or you just build your gpu opts and call `nvidia.WithGPUs(nvidiaOpts)(ctx, nil, nil, spec)`